### ISSUE TITLE
Value.joinInstances not possible for i8:0 and {i8:0,i8:127} using numeric.this and integer.this

### ISSUE THREAD

**Issue by maxteufel** [tokiwa-software/fuzion#1121]
Introducing this change:

```diff
diff --git a/lib/integer.fz b/lib/integer.fz
index e8cc60bc..28596a06 100644
--- a/lib/integer.fz
+++ b/lib/integer.fz
@@ -110,7 +110,7 @@ integer(I type : integer I) : numeric I is
   is
     b := from_u32 base
 
-    if thiz.sign < 0
+    if integer.this.sign ⩻ 0
       # there could be an overflow on negation
       match -? thiz
         v I => "-" + v.as_string base
diff --git a/lib/numeric.fz b/lib/numeric.fz
index 34e9afbe..1314325b 100644
--- a/lib/numeric.fz
+++ b/lib/numeric.fz
@@ -117,7 +117,7 @@ numeric(redef T type : numeric T) : has_hash, has_total_order, numerics T is
   infix **^(other T) T is abstract
 
 
-  sign => if thiz ≟ zero then 0 else if thiz ⩼ zero then 1 else -1
+  sign => if numeric.this ≟ numeric.this.type.zero then 0 else if thiz ⩼ zero then 1 else -1
 
   abs => if sign ≥ 0 then thiz else -thiz
```

(note that both changes are necessary, dropping one of them makes this problem disappear)

Then

```
ex_odd3 is
  i1 i8 := 0
  i2 i8 := 127
  x := "$i1 $i2"
```

results in this error:

```
error 1: java.lang.Error: Value.joinInstances not possible for i8:0 and {i8:0,i8:127}
	at dev.flang.fuir.analysis.dfa.Value.joinInstances(Value.java:267)
	at dev.flang.fuir.analysis.dfa.NumericValue.joinInstances(NumericValue.java:221)
	at dev.flang.fuir.analysis.dfa.Value.join(Value.java:256)
	at dev.flang.fuir.analysis.dfa.Instance.setField(Instance.java:133)
	at dev.flang.fuir.analysis.dfa.DFA.analyze(DFA.java:869)
	at dev.flang.fuir.analysis.dfa.DFA.iteration(DFA.java:845)
	at dev.flang.fuir.analysis.dfa.DFA.findFixPoint(DFA.java:808)
	at dev.flang.fuir.analysis.dfa.DFA.dfa(DFA.java:790)
	at dev.flang.fuir.analysis.dfa.DFA.new_fuir(DFA.java:719)
	at dev.flang.be.c.C.<init>(C.java:495)
	at dev.flang.tools.Fuzion$Backend$2.process(Fuzion.java:145)
	at dev.flang.tools.Fuzion$Backend.processFrontEnd(Fuzion.java:368)
	at dev.flang.tools.Fuzion.lambda$parseArgsForBackend$3(Fuzion.java:832)
	at dev.flang.tools.Tool.lambda$run$0(Tool.java:154)
	at dev.flang.util.Errors.runAndExit(Errors.java:704)
	at dev.flang.tools.Tool.run(Tool.java:154)
	at dev.flang.tools.Fuzion.main(Fuzion.java:481)


*** fatal errors encountered, stopping.
```

```
ex_odd3 is
  i1 i8 := 0
  i2 i8 := 127
  x := "$i1"
```

results in C compiler errors.

**Comment by fridis** [comment]
After fixes in DFA (#1142) and the C backend (#1145), this still fails, it looks as if an `Unbox` is missing for values passed to `i8.type.lteq`.

**Comment by michaellilltokiwa** [comment]
works now, with this updated patch applied:

```
diff --git a/lib/integer.fz b/lib/integer.fz
index b233f8ce0..56de53cdc 100644
--- a/lib/integer.fz
+++ b/lib/integer.fz
@@ -111,7 +111,7 @@ module:public integer : numeric is
   =>
     b := integer.this.type.from_u32 base
 
-    if integer.this.sign < 0
+    if integer.this.sign ⩻ 0
       # there could be an overflow on negation
       match -? integer.this
         v integer.this => "-" + v.as_string base
diff --git a/lib/numeric.fz b/lib/numeric.fz
index b9aa6e269..572e1cdb9 100644
--- a/lib/numeric.fz
+++ b/lib/numeric.fz
@@ -112,8 +112,7 @@ module:public numeric : property.hashable, property.orderable is
   public infix *^ (other numeric.this) numeric.this => numeric.this * other
   public infix **^(other numeric.this) numeric.this => abstract
 
-
-  public sign => if numeric.this = numeric.this.type.zero then 0 else if numeric.this > numeric.this.type.zero then 1 else -1
+  public sign => if numeric.this ≟ numeric.this.type.zero then 0 else if numeric.this ⩼ numeric.this.type.zero then 1 else -1
 
   public abs => if sign ≥ 0 then numeric.this else -numeric.this
 

```
